model/tasks/ChangePermissionsTask.php

```diff
@@ -43,32 +44,101 @@
  */
 class ChangePermissionsTask extends AbstractAction implements TaskAwareInterface, JsonSerializable
 {
-    use ServiceManagerAwareTrait;
     use TaskAwareTrait;
     use OntologyAwareTrait;
 
-    public const PARAM_RECURSIVE = 'recursive';
     public const PARAM_RESOURCE = 'resource';
     public const PARAM_PRIVILEGES = 'privileges';
+    public const PARAM_RECURSIVE = 'recursive';
+    public const PARAM_RECURSIVE_CLASS = 'recursive_class';
```

Comment:
I think we should not use the `RECURSIVE` name here, cause this is not recursive, but only `nested`. Maybe use `nested_resources` to be consistent with the other places?

Reply:
`public const PARAM_RECURSIVE = 'recursive';` is a legacy constant/parameter (was in line 50, I've just moved it down), I'd prefer to keep it.

Reply:
I am talking about the use of `RECURSIVE` word used for the new parameter `PARAM_RECURSIVE_CLASS` (which is not recursive.

```suggestion
    public const PARAM_NESTED_RESOURCES = 'nested_resources';
```

Reply:
The parameter is used from the controller to create the initial task.

Then, when we receive it here, we create new tasks with `PARAM_RECURSIVE=false` and with `PARAM_RECURSIVE_CLASS=true` (i.e. we turn one recursive task into n subclass tasks).

Reply:
Ok, got it: Rename `PARAM_RECURSIVE_CLASS` because that is a call to change the "resources nested under the given class". I'll do that.

Reply:
39969b3

Reply:
Resolving this thread to ease tracking things that are still pending, please reopen if you thing there's something else missing to be done related to this :)

Reply:
Thanks!!